addon/webextension/sitehelper.js

```diff
@@ -0,0 +1,41 @@
+/* globals catcher, callBackground */
+/** This is a content script added to all screenshots.firefox.com pages, and allows the site to
+    communicate with the add-on */
+
+window.sitehelper = (function () {
+
+  catcher.registerHandler((errorObj) => {
+    callBackground("reportError", errorObj);
+  });
+
+
+  function sendCustomEvent(name, detail) {
+    if (typeof detail == "object") {
+      // Note sending an object can lead to security problems, while a string
+      // is safe to transfer:
+      detail = JSON.stringify(detail);
```

Comment:
You would be better off with sendMessage than custom events, since messages are structured cloned by default. However, if you need to use custom events, you can use `cloneInto` to make the object accessible to content.